Enterprise-CMCS/macpro-onemac
# Show data filtering options as chips above PortalTable

Story: https://qmacbis.atlassian.net/browse/OY2-23426

### Details

Adds a tray of chips displaying filtering options applied (additive) or removed (subtractive) from the table.

Subtractive filter types are filters that come with all options pre-applied, and allow the user to reduce the number of applied options. Examples: Type, Status

Additive filter types are filters that come with no options pre-applied, and allow the user to reduce visible data by adding a condition. Examples: Territory, time-based filters

Filters are displayed as either `Hidden [Filter]: [Value]` or `[Filter]: [Value]` depending on if they're subtractive or additive.

<img width="1920" alt="Screen Shot 2023-07-20 at 3 12 46 PM" src="https://github.com/Enterprise-CMCS/macpro-onemac/assets/4022304/16a4a209-f0e0-46de-9292-5ec8aef0e65b">

### Changes

- Adds Chip, FilterChipTray
- Enumerates column IDs

### Implementation Notes

- One way we can improve the current implementation is to refactor the UI-friendly names for statuses out of `app-api` and into `common`

### Test Plan

1. Login as a State user
2. Navigate to the Dashboard
3. Apply filters by clicking `Filter` and adding/removing options for each
4. Ensure the filter chips show the filters you have applied
5. Ensure each filter can be removed with the `X` icon next to its value
6. Repeat steps 3-5 on the Waivers tab
7. Login as a CMS user
8. Repeat steps 2-6
9. Finally, ensure refreshing and log-out behaviors are still to persist and wipe applied filters respectively
